gst/nnstreamer/elements/gsttensor_trainer.c

```diff
@@ -402,7 +402,8 @@ gst_tensor_trainer_set_property (GObject * object, guint prop_id,
       if (trainer->ready_to_complete_training == g_value_get_boolean (value))
         break;
       trainer->ready_to_complete_training = g_value_get_boolean (value);
-      if (trainer->ready_to_complete_training)
+      if (trainer->ready_to_complete_training == TRUE
+          && trainer->is_training_complete == FALSE)
```

Comment:
Simple Question)
There is lots of flag variables such as `is_training_complete`, `ready_to_complete_training` and `is_epoch_complete`. Could it make a single state variable? In my opinion, lots of boolean variables make hard to understand the code.

Reply:
All variables are required.